assets/javascripts/discourse/widgets/vote-button.js

```diff
@@ -71,24 +72,30 @@ export default createWidget("vote-button", {
   },
 
   click() {
-    if (!this.currentUser) {
-      this.sendWidgetAction("showLogin");
-      cookie("destination_url", window.location.href, { path: "/" });
-      return;
-    }
-    if (
-      !this.attrs.closed &&
-      this.parentWidget.state.allowClick &&
-      !this.attrs.user_voted &&
-      !this.currentUser.votes_exceeded
-    ) {
-      this.parentWidget.state.allowClick = false;
-      this.parentWidget.state.initialVote = true;
-      this.sendWidgetAction("addVote");
-    }
-    if (this.attrs.user_voted || this.currentUser.votes_exceeded) {
-      document.querySelector(".vote-options").classList.toggle("hidden");
-    }
+    applyBehaviorTransformer(
+      "topic-vote-button-click",
+      () => {
+        if (!this.currentUser) {
+          this.sendWidgetAction("showLogin");
+          cookie("destination_url", window.location.href, { path: "/" });
+          return;
+        }
+        if (
+          !this.attrs.closed &&
+          this.parentWidget.state.allowClick &&
+          !this.attrs.user_voted &&
+          !this.currentUser.votes_exceeded
+        ) {
+          this.parentWidget.state.allowClick = false;
+          this.parentWidget.state.initialVote = true;
+          this.sendWidgetAction("addVote");
+        }
+        if (this.attrs.user_voted || this.currentUser.votes_exceeded) {
+          document.querySelector(".vote-options").classList.toggle("hidden");
+        }
+      },
+      { currentUser: this.currentUser, attrs: this.attrs }
```

Comment:
Passing attrs means that this API is being tied to the widget implementation. Would be better if we can pass specific things, which will still make sense once this is converted to be a glimmer component.